packages/x-components/src/x-modules/scroll/components/scroll.vue

```diff
@@ -30,92 +29,124 @@
    *
    * @public
    */
-  @Component({
-    mixins: [xComponentMixin(scrollXModule)],
-    components: { BaseScroll }
-  })
-  export default class Scroll extends Vue {
-    /**
-     * Id to identify the component.
-     *
-     * @public
-     */
-    @Prop({ default: MainScrollId })
-    public id!: string;
-
-    /**
-     * Emits the `UserScrolled` event.
-     *
-     * @param position - The new position of scroll.
-     * @internal
-     */
-    protected emitScroll(position: number): void {
-      this.$x.emit('UserScrolled', position, this.createEventMetadata());
-    }
+  export default defineComponent({
+    name: 'Scroll',
+    xModule: scrollXModule.name,
+    components: { BaseScroll },
+    props: {
+      /**
+       * Id to identify the component.
+       *
+       * @public
+       */
+      id: {
+        type: String,
+        default: MainScrollId
+      }
+    },
+    setup(props) {
+      type ElementRef = {
+        $el: HTMLElement;
+      };
 
-    /**
-     * Emits the `UserChangedScrollDirection` event when the scrolling direction has changed.
-     *
-     * @param direction - The new direction of scroll.
-     * @internal
-     */
-    protected emitScrollDirectionChange(direction: ScrollDirection): void {
-      this.$x.emit('UserChangedScrollDirection', direction, this.createEventMetadata());
-    }
+      const $x = use$x();
 
-    /**
-     * Emits the 'UserReachedScrollStart' event when the user reaches the start.
-     *
-     * @param isAtStart - A boolean indicating if the scroll is at the ending position.
-     * @internal
-     */
-    protected emitScrollAtStart(isAtStart: boolean): void {
-      this.$x.emit('UserReachedScrollStart', isAtStart, this.createEventMetadata());
-    }
+      const isElementRef = (value: any): value is ElementRef => {
```

Comment:
is this really necessary after all? Since we are using the ref over the `BaseScroll` I would image that we are always getting a ref to the vue component. 🤔